src/matlab2tikz.m

```diff
@@ -1000,12 +1049,116 @@ case guitypes()
 end
 end
 % ==============================================================================
+function m2t = getPlotyyReferences(m2t,axisHandle)
+% Retrieve references to legend entries of the main plotyy axis
+%
+% A plotyy plot has a main and a secondary axis. The legend is associated
+% with the main axis and hence m2t will only include the legend entries 
+% that belong to the \axis[] that has a legend. 
+%
+% One way to include the legend entries from the secondary axis (in the 
+% same legend) is to first label the \addplot[] and then reference them. 
+% See https://tex.stackexchange.com/questions/42697/42752#42752
+%
+% However, in .tex labels should come before they are referenced. Hence, 
+% we actually label the legend entries from the main axis and swap the 
+% legendhandle to the secondary axis. 
+%
+% The legend will not be plotted with the main \axis[] and the labelled
+% legend entries will be skipped until the secondary axis. Then, they will
+% be listed before any legend entry from the secondary axis.
+
+% Retrieve legend handle
+if isAxisMain(axisHandle)
+    legendHandle = m2t.axesContainers{end}.LegendHandle;
+else
+    legendHandle = getAssociatedLegend(m2t,getPlotyyPeer(axisHandle));
+    m2t.axesContainers{end}.LegendHandle = legendHandle;
+end
+
+% Not a plotyy axis or no legend
+if ~isAxisPlotyy(axisHandle) || isempty(legendHandle)
+    m2t.axesContainers{end}.PlotyyReferences = [];
+
+elseif isAxisMain(axisHandle)
+    % Mark legend entries of the main axis for labelling
+    legendEntries = m2t.axesContainers{end}.LegendEntries;
+    ancAxes       = ancestor(legendEntries,'axes');
+    idx           = ismember(double([ancAxes{:}]), axisHandle);
+    m2t.axesContainers{end}.PlotyyReferences = legendEntries(idx);
+
+    % Ensure no legend is created on the main axis
+    m2t.axesContainers{end}.LegendHandle = [];
+else
+    % Get legend entries associated to secondary plotyy axis. We can do
+    % this because we took the legendhandle from the peer (main axis)
+    legendEntries = getLegendEntries(m2t);
+    ancAxes       = ancestor(legendEntries,'axes');
+    if iscell(ancAxes)
+        ancAxes = [ancAxes{:}];
+    end
+    idx = ismember(double(ancAxes), axisHandle);
+    m2t.axesContainers{end}.LegendEntries = legendEntries(idx);
+
+    % Recover referenced legend entries of the main axis
+    m2t.axesContainers{end}.PlotyyReferences = legendEntries(~idx);
+end
+end
+% ==============================================================================
+function bool = isAxisMain(h)
+% Check if it is the main axis e.g. in a plotyy plot
+
+if ~isAxisPlotyy(h)
```

Comment:
Just a minor thing: I think it makes more sense to do:
```matlab
bool = true;
if isAxisPlotyy(h)
    ...
end
```

than (the current)
```matlab
if ~isAxisPlotyy(h)
    bool = true;
    return
end

...
```

as the latter makes the control flow more obscure IMHO (for exceptional cases (e.g. invisible lines, empty plots, ...), early `return`s can sometimes be advantageous, but I don't think that is the case here).